scipy/special/_basic.py

```diff
@@ -2953,6 +2956,139 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def stirling2(N, K, exact=True):
```

Comment:
For new functions, we should directly distinguish between positional & keyword-only arguments, i.e.
```suggestion
def stirling2(N, K, *, exact=True):
```